frontend/alert/components/AutoComplete.tsx

```diff
@@ -283,6 +243,29 @@ const AutoComplete = ({ defaultValue="", onValueChange, setTimeline, disabled })
             .map((suggestion) => suggestion.section_id.toLowerCase())
             .indexOf(value.toLowerCase());
 
+    /**
+     * Returns suggested suggestion grouped by course
+     * @return suggestions
+     */
+    const groupedSuggestions = suggestions
+        .sort((a, b) => a.section_id.localeCompare(b.section_id))
+        .reduce((res, obj) => {
+            const [courseName, midNum, endNum] = obj.section_id.split("-");
+            const { activity } = obj;
+
+            if (res[`${courseName}-${midNum}`]) {
+                if (res[`${courseName}-${midNum}`][activity]) {
+                    res[`${courseName}-${midNum}`][activity].push(obj);
+                } else {
+                    res[`${courseName}-${midNum}`][activity] = [obj];
+                }
```

Comment:
Could just be ``res[`${courseName}-${midNum}`][activity].push(obj);`` (eliminating the else case) since we know `res` values are always initialized to the form `{ [activity]: [obj] }`, right?

Reply:
Hmm, I'm basically grouping all the sections by its course and then grouping by its activity (subgroups) within each group. Something like this:

`CIS-120:
     LEC: [{CIS-120-100, CIS-120-101}]
     REC: [{CIS-120-110, CIS-120-111}]
`

`res[`${courseName}-${midNum}`] = { [activity]: [obj] }` would create a new group (for a new course) and  `res[`${courseName}-${midNum}`][activity] = [obj]` would create a new subgroup (for a new activity that current isnt a subgroup) within each group. Just having `res[`${courseName}-${midNum}`][activity].push(obj)` wouldn't work when its a new activity that isn't a subgroup yet right? I could totally be wrong lol

Is there a better way of grouping the sections into this format?

Reply:
Oh my bad, you're totally right.